planner/core/planbuilder.go

```diff
@@ -304,7 +304,11 @@ func (hch *handleColHelper) pushMap(m map[int64][]*expression.Column) {
 }
 
 func (hch *handleColHelper) mergeAndPush(m1, m2 map[int64][]*expression.Column) {
-	newMap := make(map[int64][]*expression.Column)
+	mapMinSize := len(m1)
+	if len(m2) > len(m1) {
```

Comment:
We can use `mathutil.Min` function to simplify the code.

Reply:
Addressed